flink-staging/flink-streaming/flink-streaming-core/src/test/java/org/apache/flink/streaming/api/functions/sink/SocketClientSinkTest.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.api.functions.sink;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.net.Socket;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.util.serialization.SerializationSchema;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.net.ServerSocket;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Tests for the {@link org.apache.flink.streaming.api.functions.sink.SocketClientSink}.
+ */
+public class SocketClientSinkTest{
+
+	final AtomicReference<Throwable> error = new AtomicReference<Throwable>();
+	private final String host = "127.0.0.1";
+	private int port;
+	private String value;
+
+	public Thread t;
+
+	public SocketClientSinkTest() {
+	}
+
+	@Test
+	public void testSocketSink() throws Exception{
+		value = "";
+		ServerSocket server = new ServerSocket(0);
+		port = server.getLocalPort();
+
+		new Thread(new Runnable() {
+			@Override
+			public void run() {
+				t = Thread.currentThread();
+				SerializationSchema<String, byte[]> simpleSchema = new SerializationSchema<String, byte[]>() {
+					@Override
+					public byte[] serialize(String element) {
+						return element.getBytes();
+					}
+				};
+
+				try {
+					SocketClientSink<String> simpleSink = new SocketClientSink<String>(host, port, simpleSchema, 0);
+					simpleSink.open(new Configuration());
+					simpleSink.invoke("testSocketSinkInvoke");
+					simpleSink.close();
+				} catch (Exception e){
+					error.set(e);
+				}
+			}
+		}).start();
+
+		Socket sk = server.accept();
+		BufferedReader rdr = new BufferedReader(new InputStreamReader(sk
+				.getInputStream()));
+		value = rdr.readLine();
+
+		t.join();
+		server.close();
+
+		if (error.get() != null) {
+			Throwable t = error.get();
```

Comment:
I think it's good to set the error to `null` again. Otherwise all subsequent tests will fail with the same error message which is, however, completely unrelated.